test/base/SpecBase.scala

```diff
@@ -146,25 +148,67 @@ trait SpecBase
+      val taxRegisteredInEuSummaryRow = TaxRegisteredInEuSummary.row(answers, mode)
+      val euDetailsSummaryRow = EuDetailsSummary.checkAnswersRow(answers, mode)
+      val isOnlineMarketplaceSummaryRow = IsOnlineMarketplaceSummary.row(answers, mode)
+      val hasWebsiteSummaryRow = HasWebsiteSummary.row(answers, mode)
+      val websiteSummaryRow = WebsiteSummary.checkAnswersRow(answers, mode)
+      val businessContactDetailsContactNameSummaryRow = BusinessContactDetailsSummary.rowContactName(answers, mode)
+      val businessContactDetailsTelephoneSummaryRow = BusinessContactDetailsSummary.rowTelephoneNumber(answers, mode)
+      val businessContactDetailsEmailSummaryRow= BusinessContactDetailsSummary.rowEmailAddress(answers, mode)
+      val bankDetailsAccountNameSummaryRow = BankDetailsSummary.rowAccountName(answers, mode)
+      val bankDetailsBicSummaryRow = BankDetailsSummary.rowBIC(answers, mode)
+      val bankDetailsIbanSummaryRow = BankDetailsSummary.rowIBAN(answers, mode)
+
       Seq(
-        new HasTradingNameSummary().row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        HasMadeSalesSummary.row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        IsPlanningFirstEligibleSaleSummary.row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        Some(commencementDateSummary),
-        PreviouslyRegisteredSummary.row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        TaxRegisteredInEuSummary.row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        EuDetailsSummary.checkAnswersRow(answers, mode),
-        IsOnlineMarketplaceSummary.row(answers, mode),
-        HasWebsiteSummary.row(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        BusinessContactDetailsSummary.rowContactName(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        BusinessContactDetailsSummary.rowTelephoneNumber(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        BusinessContactDetailsSummary.rowEmailAddress(answers, mode),
-        BankDetailsSummary.rowAccountName(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        BankDetailsSummary.rowBIC(answers, mode).map(_.withCssClass("govuk-summary-list__row--no-border")),
-        BankDetailsSummary.rowIBAN(answers, mode)
+        hasTradingNameSummaryRow.map(sr =>
+        if(tradingNameSummaryRow.isDefined) {
+          sr.withCssClass("govuk-summary-list__row--no-border")
+        } else {
+          sr
+        }),
+        tradingNameSummaryRow,
+        hasMadeSalesSummaryRow.map(_.withCssClass("govuk-summary-list__row--no-border")),
+        isPlanningFirstEligibleSaleSummaryRow.map(_.withCssClass("govuk-summary-list__row--no-border")),
+        commencementDateSummaryRow,
+        previouslyRegisteredSummaryRow.map(sr =>
```

Comment:
can we use `{` on the map as it would improve indentation and therefore readability